### ISSUE TITLE
Help getting started with running a package in WSL

### ISSUE THREAD

**Issue by matthias-github** [pythonnet/pythonnet#1639]
### Environment

-   Pythonnet version:  3.0.0dev
-   Python version: 3.8.10
-   Operating System: Ubuntu on WSL
-   .NET Runtime: .NET 5

### Details
- First, I am a complete noob when it comes to .NET. I simply know that it has some function I need, and from python I can access these if I manage to AddReference to these dll. With that, we have built a python code that uses:
1. Microsoft.Data.SqlClient
2. Microsoft.AnalysisServices.Tabular
3. Microsoft.AnalysisServices.Tabular.Json
4. Microsoft.AnalysisServices.AdomdClient

The code works when I simply list out these dll (plus a few more) into a folder in Windows, and add reference to them using clr.AddReference.  

My goal is to port over the code to Linux. I managed to get pythonnet installed/running on WSL. 

-   What commands did you run to trigger this issue? If you can provide a
    [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
    this will help us understand the issue.

This is how I set-up my system
```
sudo apt-get update

# install dotnet
wget https://packages.microsoft.com/config/ubuntu/20.04/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
sudo dpkg -i packages-microsoft-prod.deb
sudo apt-get install -y apt-transport-https
sudo apt-get update
sudo apt-get install -y dotnet-sdk-5.0

# install mono/things that pythonnet seems to need
sudo apt install mono-devel
sudo apt-get install clang 
sudo apt-get install libglib2.0-dev 

# install python related stuff
sudo apt-get install python3-pip python3.8-venv
sudo apt-get install git

# create and load venv
python3 -m venv venv
source venv/bin/activate

# install pythonnet
venv/bin/pip install pycparser
venv/bin/pip install git+https://github.com/pythonnet/pythonnet
```

-   When this is done, I can run pythonnet successfully

```
>>> import clr
>>> clr.AddReference('System.Data')
<System.Reflection.RuntimeAssembly object at 0x7f7f7056cc00>
>>> from System.Data import DataTable
>>> assert DataTable
>>>
```

From here on, I am totally stuck. If I try to "dump" the same dll into a folder and AddReference, I can add the assembly. But when I load it, I get something a complaint:
```
>>>
 clr.AddReference('Microsoft.Data.SqlClient')
<System.Reflection.RuntimeAssembly object at 0x7f7f67858200>
>>> from Microsoft.Data.SqlClient import SqlConnection
```

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 657, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 556, in module_from_spec
  File "<string>", line 15, in create_module
System.TypeLoadException: Could not resolve type with token 01000112 from typeref (expected class 'System.Security.Cryptography.ECDiffieHellmanCng' in assembly 'System.Core, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089')
  at (wrapper managed-to-native) System.Reflection.MonoMethodInfo.get_method_info(intptr,System.Reflection.MonoMethodInfo&)
  at System.Reflection.MonoMethodInfo.GetMethodInfo (System.IntPtr handle) [0x00000] in <12b418a7818c4ca0893feeaaf67f1e7f>:0
  at System.Reflection.MonoMethodInfo.GetDeclaringType (System.IntPtr handle) [0x00000] in <12b418a7818c4ca0893feeaaf67f1e7f>:0
  at System.Reflection.RuntimeMethodInfo.get_DeclaringType () [0x00000] in <12b418a7818c4ca0893feeaaf67f1e7f>:0
  at Python.Runtime.ClassManager.GetClassInfo (System.Type type) [0x00033] in <20dac50b00a54887a15166c893d26620>:0
  at Python.Runtime.ClassManager.InitClassBase (System.Type type, Python.Runtime.ClassBase impl, Python.Runtime.PyType pyType) [0x00000] in <20dac50b00a54887a15166c893d26620>:0
  at Python.Runtime.ReflectedClrType.GetOrCreate (System.Type type) [0x0003d] in <20dac50b00a54887a15166c893d26620>:0
  at Python.Runtime.ClassManager.GetClass (System.Type type) [0x00000] in <20dac50b00a54887a15166c893d26620>:0
  at Python.Runtime.ModuleObject.GetAttribute (System.String name, System.Boolean guess) [0x00099] in <20dac50b00a54887a15166c893d26620>:0
  at Python.Runtime.ModuleObject.LoadNames () [0x0004b] in <20dac50b00a54887a15166c893d26620>:0
  at Python.Runtime.ImportHook.Import (System.String modname) [0x00092] in <20dac50b00a54887a15166c893d26620>:0
  at Python.Runtime.CLRModule._load_clr_module (Python.Runtime.PyObject spec) [0x00022] in <20dac50b00a54887a15166c893d26620>:0
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0007c] in <12b418a7818c4ca0893feeaaf67f1e7f>:0
>>>
```

I tried following basic online tutorials of making a dotnet application via addpackage etc,, and I was able to run this::

```
using System;
using Microsoft.Data.SqlClient;

namespace foo
{
    class Program
    {
        
        static void Main(string[] args)
        {
            using var cnn = new SqlConnection("Server=localhost; Initial Catalog=admin; Integrated Security=true; pooling=false;");
            Console.WriteLine("Hello World!");
        }
    }
}
```

In other words, I simply do not how to get it running and I was hoping to get some help.

**Comment by lostmsu** [comment]
This is not a support forum

**Comment by CrasCris** [comment]
> ### Environment
> * Pythonnet version:  3.0.0dev
> * Python version: 3.8.10
> * Operating System: Ubuntu on WSL
> * .NET Runtime: .NET 5
> 
> ### Details
> * First, I am a complete noob when it comes to .NET. I simply know that it has some function I need, and from python I can access these if I manage to AddReference to these dll. With that, we have built a python code that uses:
> 
> 1. Microsoft.Data.SqlClient
> 2. Microsoft.AnalysisServices.Tabular
> 3. Microsoft.AnalysisServices.Tabular.Json
> 4. Microsoft.AnalysisServices.AdomdClient
> 
> The code works when I simply list out these dll (plus a few more) into a folder in Windows, and add reference to them using clr.AddReference.
> 
> My goal is to port over the code to Linux. I managed to get pythonnet installed/running on WSL.
> 
> * What commands did you run to trigger this issue? If you can provide a
>   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
>   this will help us understand the issue.
> 
> This is how I set-up my system
> 
> ```
> sudo apt-get update
> 
> # install dotnet
> wget https://packages.microsoft.com/config/ubuntu/20.04/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
> sudo dpkg -i packages-microsoft-prod.deb
> sudo apt-get install -y apt-transport-https
> sudo apt-get update
> sudo apt-get install -y dotnet-sdk-5.0
> 
> # install mono/things that pythonnet seems to need
> sudo apt install mono-devel
> sudo apt-get install clang 
> sudo apt-get install libglib2.0-dev 
> 
> # install python related stuff
> sudo apt-get install python3-pip python3.8-venv
> sudo apt-get install git
> 
> # create and load venv
> python3 -m venv venv
> source venv/bin/activate
> 
> # install pythonnet
> venv/bin/pip install pycparser
> venv/bin/pip install git+https://github.com/pythonnet/pythonnet
> ```
> 
> * When this is done, I can run pythonnet successfully
> 
> ```
> >>> import clr
> >>> clr.AddReference('System.Data')
> <System.Reflection.RuntimeAssembly object at 0x7f7f7056cc00>
> >>> from System.Data import DataTable
> >>> assert DataTable
> >>>
> ```
> 
> From here on, I am totally stuck. If I try to "dump" the same dll into a folder and AddReference, I can add the assembly. But when I load it, I get something a complaint:
> 
> ```
> >>>
>  clr.AddReference('Microsoft.Data.SqlClient')
> <System.Reflection.RuntimeAssembly object at 0x7f7f67858200>
> >>> from Microsoft.Data.SqlClient import SqlConnection
> ```
> 
> ```
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "<frozen importlib._bootstrap>", line 991, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
>   File "<frozen importlib._bootstrap>", line 657, in _load_unlocked
>   File "<frozen importlib._bootstrap>", line 556, in module_from_spec
>   File "<string>", line 15, in create_module
> System.TypeLoadException: Could not resolve type with token 01000112 from typeref (expected class 'System.Security.Cryptography.ECDiffieHellmanCng' in assembly 'System.Core, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089')
>   at (wrapper managed-to-native) System.Reflection.MonoMethodInfo.get_method_info(intptr,System.Reflection.MonoMethodInfo&)
>   at System.Reflection.MonoMethodInfo.GetMethodInfo (System.IntPtr handle) [0x00000] in <12b418a7818c4ca0893feeaaf67f1e7f>:0
>   at System.Reflection.MonoMethodInfo.GetDeclaringType (System.IntPtr handle) [0x00000] in <12b418a7818c4ca0893feeaaf67f1e7f>:0
>   at System.Reflection.RuntimeMethodInfo.get_DeclaringType () [0x00000] in <12b418a7818c4ca0893feeaaf67f1e7f>:0
>   at Python.Runtime.ClassManager.GetClassInfo (System.Type type) [0x00033] in <20dac50b00a54887a15166c893d26620>:0
>   at Python.Runtime.ClassManager.InitClassBase (System.Type type, Python.Runtime.ClassBase impl, Python.Runtime.PyType pyType) [0x00000] in <20dac50b00a54887a15166c893d26620>:0
>   at Python.Runtime.ReflectedClrType.GetOrCreate (System.Type type) [0x0003d] in <20dac50b00a54887a15166c893d26620>:0
>   at Python.Runtime.ClassManager.GetClass (System.Type type) [0x00000] in <20dac50b00a54887a15166c893d26620>:0
>   at Python.Runtime.ModuleObject.GetAttribute (System.String name, System.Boolean guess) [0x00099] in <20dac50b00a54887a15166c893d26620>:0
>   at Python.Runtime.ModuleObject.LoadNames () [0x0004b] in <20dac50b00a54887a15166c893d26620>:0
>   at Python.Runtime.ImportHook.Import (System.String modname) [0x00092] in <20dac50b00a54887a15166c893d26620>:0
>   at Python.Runtime.CLRModule._load_clr_module (Python.Runtime.PyObject spec) [0x00022] in <20dac50b00a54887a15166c893d26620>:0
>   at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
>   at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0007c] in <12b418a7818c4ca0893feeaaf67f1e7f>:0
> >>>
> ```
> 
> I tried following basic online tutorials of making a dotnet application via addpackage etc,, and I was able to run this::
> 
> ```
> using System;
> using Microsoft.Data.SqlClient;
> 
> namespace foo
> {
>     class Program
>     {
>         
>         static void Main(string[] args)
>         {
>             using var cnn = new SqlConnection("Server=localhost; Initial Catalog=admin; Integrated Security=true; pooling=false;");
>             Console.WriteLine("Hello World!");
>         }
>     }
> }
> ```
> 
> In other words, I simply do not how to get it running and I was hoping to get some help.

Did you find out how to use it ? , i try to connect with SSAS but i got bunch of errors in Linux
